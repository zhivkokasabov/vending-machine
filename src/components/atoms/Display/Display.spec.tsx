import { render } from "@testing-library/react";

import { Display } from ".";

import * as coinManager from "providers/CoinManager.provider";
import * as codeProvider from "providers/Code.provider";
import * as messageProvider from "providers/Message.provider";

jest.mock("providers/CoinManager.provider");
jest.mock("providers/Code.provider");
jest.mock("providers/Message.provider");

describe("CoinSlotPanel tests", () => {
  const total = 12;
  const code = "code";
  const message = "message";

  const renderComponent = () => render(
     <Display />
  );

  beforeEach(() => {
    jest.spyOn(coinManager, 'useCoinManager').mockReturnValue({
      insertCoin: jest.fn(),
      reset: jest.fn(),
      total
    });
    jest.spyOn(codeProvider, 'useCode').mockReturnValue({
      append: jest.fn(),
      resetCode: jest.fn(),
      code
    });
    jest.spyOn(messageProvider, 'useMessage').mockReturnValue({
      reset: jest.fn(),
      set: jest.fn(),
      message
    });
  });

  it("should render", () => {
    const { asFragment } = renderComponent();

    expect(asFragment()).toMatchSnapshot();
  });
});
