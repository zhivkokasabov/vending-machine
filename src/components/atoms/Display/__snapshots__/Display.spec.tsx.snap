// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinSlotPanel tests should render 1`] = `
<DocumentFragment>
  <div
    class="bg-black rounded p-4 text-right text-white undefined"
    data-testid="display"
  >
    <p
      class="text-4xl text-sky-600 font-bold mb-2"
    >
      12
    </p>
    <p
      class="h-6 text-2xl max-w-40 text-ellipsis overflow-hidden mb-2"
    >
      code
    </p>
    <p
      class="h-10 text-sm"
    >
      message
    </p>
  </div>
</DocumentFragment>
`;
